DiscordChatExporter.Core/Exporting/ExportAssetDownloader.cs

```diff
@@ -101,12 +102,24 @@ await using (var output = File.Create(filePath))
 
 internal partial class ExportAssetDownloader
 {
-    private static string GetUrlHash(string url) =>
-        SHA256
+    private static string GetUrlHash(string url)
+    {
+        // Strip out ex, is and hm query params to create a consistent hash
+        var uriBuilder = new UriBuilder(url);
+        var query = HttpUtility.ParseQueryString(uriBuilder.Query);
+        query.Remove("ex");
+        query.Remove("is");
+        query.Remove("hm");
+        uriBuilder.Query = query.ToString();
+        url = uriBuilder.ToString();
+        url = url.Replace(":443", "");
+
+        return SHA256
```

Comment:
Let's do it only for Discord CDN URLs. These parameters can have some other meaning on other hosts.